src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -65,33 +74,72 @@ protected override bool ValidateParameters()
         {
             _createPDBCommand = CompilationEntry.GetMetadata(MetadataKeys.CreatePDBCommand);
 
-            if (CrossgenTool == null && Crossgen2Tool == null)
-            {
-                return false;
-            }
             if (IsPdbCompilation && CrossgenTool == null)
             {
-                // We need the crossgen tool for now to emit native symbols. Crossgen2 does not yet support this feature
+                // PDB compilation is a step specific to Crossgen1 and 5.0 Crossgen2
+                // which didn't support PDB generation. 6.0  Crossgen2 produces symbols
+                // directly during native compilation.
+                Log.LogError(Strings.CrossgenToolMissingInPDBCompilationMode);
                 return false;
             }
 
-            if (CrossgenTool != null)
+            if (UseCrossgen2)
             {
-                if (!File.Exists(CrossgenTool.ItemSpec) || !File.Exists(CrossgenTool.GetMetadata(MetadataKeys.JitPath)))
+                // We expect JitPath to be set for .NET 5 and {TargetOS, TargetArch} to be set for .NET 6 and later
+                string jitPath = Crossgen2Tool.GetMetadata(MetadataKeys.JitPath);
+                if (Crossgen2Tool == null || string.IsNullOrEmpty(Crossgen2Tool.ItemSpec))
+                {
+                    Log.LogError(Strings.Crossgen2ToolMissingWhenUseCrossgen2IsSet);
+                    return false;
+                }
+                if (!File.Exists(Crossgen2Tool.ItemSpec))
                 {
+                    Log.LogError(Strings.Crossgen2ToolExecutableNotFound, Crossgen2Tool.ItemSpec);
                     return false;
                 }
+                string hostPath = DotNetHostPath;
+                if (!string.IsNullOrEmpty(hostPath) && !File.Exists(hostPath))
+                {
+                    Log.LogError(Strings.DotNetHostExecutableNotFound, hostPath);
+                    return false;
+                }
+                if (!String.IsNullOrEmpty(jitPath))
+                {
+                    if (!File.Exists(jitPath))
+                    {
+                        Log.LogError(Strings.JitLibraryNotFound, jitPath);
+                        return false;
+                    }
+                }
+                else
+                {
+                    if (string.IsNullOrEmpty(Crossgen2Tool.GetMetadata(MetadataKeys.TargetOS)))
+                    {
+                        Log.LogError(Strings.TargetOSMissingWhenJitPathNotSet);
+                        return false;
+                    }
+                    if (string.IsNullOrEmpty(Crossgen2Tool.GetMetadata(MetadataKeys.TargetArch)))
+                    {
+                        Log.LogError(Strings.TargetArchMissingWhenJitPathNotSet);
+                        return false;
+                    }
```

Comment:
We could use a single parameterized error message and pass either `'TargetOS'` or `'TargetArch'`.

Reply:
Good point, fixed in 6th commit.